### PR TITLE
feat: migrate panels components to vue 3 syntax

### PR DESCRIPTION
**Pull request by lauramargar**
EMP-424

### CODE REVIEW

File: packages/x-components/src/components/panels/__tests__/base-header-toggle-panel.spec.ts [hunk 1]
```diff
@@ -87,7 +87,7 @@ describe('testing BaseHeaderTogglePanel component', () => {
     expect(wrapper.find(getDataTestSelector('toggle-status')).text()).toBe('open');
 
     await wrapper.find(getDataTestSelector('toggle-button')).trigger('click');
-
+    wrapper.vm.$nextTick();
```
Comment (CachedaCodes) [inline packages/x-components/src/components/panels/__tests__/base-header-toggle-panel.spec.ts:90]: For me, the tests pass without this line. If this is here for something that I can't replicate, then remove the `await` in the line 89 and put it here instead.

**Review by CachedaCodes — CHANGES_REQUESTED** [review]

File: packages/x-components/src/components/panels/base-toggle-panel.vue [hunk 2]
```diff
@@ -8,34 +8,37 @@
...
+        type: Boolean,
+        required: true
+      },
+      /**
+       * Animation component that will be used to animate the panel content.
+       *
+       * @public
+       */
+      animation: {
+        type: [String, Object, Function] as PropType<string | DefineComponent>,
```
Comment (CachedaCodes) [inline packages/x-components/src/components/panels/base-toggle-panel.vue:37]: Use here the value you created in [EMP-30](https://searchbroker.atlassian.net/browse/EMP-30)

File: packages/x-components/src/components/panels/base-header-toggle-panel.vue [hunk 3]
```diff
@@ -32,52 +31,62 @@
...
+  export default defineComponent({
+    components: { BaseTogglePanel },
+    props: {
+      /**
+       * Animation component that will be used to animate the base-toggle-panel.
+       *
+       * @public
+       */
+      animation: {
+        type: [String, Object, Function] as PropType<string | DefineComponent>,
```
Comment (CachedaCodes) [inline packages/x-components/src/components/panels/base-header-toggle-panel.vue:43]: Use here the value you created in [EMP-30](https://searchbroker.atlassian.net/browse/EMP-30)

**Review by CachedaCodes — CHANGES_REQUESTED** [review]

**Review by CachedaCodes — DISMISSED** [review]

File: packages/x-components/src/utils/options-api.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+import { DefineComponent, PropType } from 'vue';
```
Comment (tiborux) [inline packages/x-components/src/utils/options-api.ts:1]: As I said in the other PR, maybe it should be in the types folder and rename it to `AnimationProp`

**Review by tiborux — APPROVED** [review]

**Pull request merged**